### PR TITLE
refactor/auth

### PR DESCRIPTION
**Pull request by riteshsp2000**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch** (left side). Also you should start *your branch* off *default branch*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] I have added necessary documentation (if appropriate)

### Description
- refactors checkPermissions function to suit for both graphql and rest APIs
- add authorization checks on user queries

❤️Thank you!

### Post merge checklist
- [ ] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

File: api/src/apollo/Event/queries.ts [hunk 1]
```diff
@@ -1,54 +1,39 @@
...
-  },
-});
-
-export const getEvents = queryField('getEvents', {
+export const event = queryField('event', {
   type: list('Event'),
-  description: `Returns a list of events depending upon the various arguments passed`,
+  description: `Returns as list of events depending upon the arguments`,
+  authorize: (_parent, args, ctx) =>
+    args.id || args.orgID ? true : checkGqlPermissions(ctx, []),
```
Comment (abhibhaw) [inline api/src/apollo/Event/queries.ts:8]: I believe getEvents should be open na?

File: api/src/apollo/Transaction/mutations.ts [hunk 2]
```diff
@@ -42,6 +45,8 @@ export const TransactionUpdateInputType = inputObjectType({
 export const updateTransaction = mutationField('updateTransaction', {
```
Comment (abhibhaw) [inline api/src/apollo/Transaction/mutations.ts:45]: this shouldn't be permitted tbh

File: api/src/apollo/User/queries.ts [hunk 3]
```diff
@@ -1,87 +1,63 @@
...
+  type: list('User'),
+  description: 'Returns a list of users depending upon the parameters passed',
+  authorize: (_parent, args, ctx) =>
+    args.id || args.uid
+      ? checkGqlPermissions(ctx, [])
+      : checkGqlPermissions(ctx, [
+          PERMISSIONS.SUPER_ADMIN,
+          PERMISSIONS.SUPER_EDITOR,
+          PERMISSIONS.SUPER_VIEWER,
+          PERMISSIONS.ORG_ADMIN,
```
Comment (abhibhaw) [inline api/src/apollo/User/queries.ts:16]: orgID is missing

**Review by abhibhaw — CHANGES_REQUESTED** [review]

File: api/src/apollo/Event/queries.ts (hunk 1 above)
Comment (riteshsp2000) [reply api/src/apollo/Event/queries.ts:8]: no it should not. If it is open for all, anyone can fetch each and every possible event (draft, other organisations, etc) which we don't really want.

File: api/src/apollo/Transaction/mutations.ts (hunk 2 above)
Comment (riteshsp2000) [reply api/src/apollo/Transaction/mutations.ts:45]: removing it

File: api/src/apollo/User/queries.ts (hunk 3 above)
Comment (riteshsp2000) [reply api/src/apollo/User/queries.ts:16]: noted

**Review by abhibhaw — APPROVED** [review]
LGTM

**Pull request merged**